.github/workflows/kit-bootstrap.yml

```diff
@@ -0,0 +1,135 @@
+            cat <<-EOF | kubectl apply -f -
+              apiVersion: kit.k8s.sh/v1alpha1
+              kind: DataPlane
+              metadata:
+                name: ${GUEST_CLUSTER_NAME}-nodes
+              spec:
+                clusterName: ${GUEST_CLUSTER_NAME} # Associated Cluster name
+                nodeCount: 2
+          EOF
+            for (( time=0; time<=1000; time+=5 )); do
+              dataplane_object=$(kubectl get dataplane ${GUEST_CLUSTER_NAME}-nodes  -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}' 2>/dev/null)
+              if [[ "$dataplane_object" == "True" ]]; then
+                echo "Dataplane objects are ready"
+                break;
+              else
+                echo "Dataplane objects are not ready"
+              fi
+              if [[ "$time" -eq 990 ]]; then
+                echo "Timeout: Dataplane objects are not ready after 10 minutes"
+                kubectl logs deploy/kit-controller -n kit 
+                exit 1
+              fi
+              sleep 5
+            done
+            for (( time=0; time<=300; time+=5 )); do
+              ready_node=$(kubectl --kubeconfig=/tmp/kubeconfig get nodes 2>/dev/null | grep -w Ready | wc -l)
+              if [[ "$ready_node" -eq 2 ]]; then
+                echo "Dataplane nodes are ready"
+                break;
+              else
+                echo "Dataplane nodes are not ready"
+              fi
+              if [[ "$time" -eq 290 ]]; then
+                echo "Timeout: Dataplane nodes are not ready after 10 minutes"
+                exit 1
+              fi
+              sleep 5
+            done
+            kubectl --kubeconfig=/tmp/kubeconfig create deployment nginx-deployment --image=nginx
+            kubectl --kubeconfig=/tmp/kubeconfig rollout status deployment/nginx-deployment
+            kubectl --kubeconfig=/tmp/kubeconfig expose deployment nginx-deployment --name=nginx-service --port=80 --target-port=80 --type=ClusterIP
+            kubectl --kubeconfig=/tmp/kubeconfig certificate approve $(kubectl --kubeconfig=/tmp/kubeconfig get csr | grep Pending | awk '{print $1}')
+            kubectl --kubeconfig=/tmp/kubeconfig run -i --rm --restart=Never curly-pod  --image=curlimages/curl --command -- curl -s 'http://nginx-service:80'
+            kubectl delete controlplane ${GUEST_CLUSTER_NAME}
+          done
+          #During the first attempt, "kitctl" was unable to delete some resources sporadically. To ensure that enough time is given for the resources to be deleted, a longer wait time of 220 seconds is added before attempting to delete the kit again.
+          #and for the remaining 4 iterations with a wait time of 60 seconds to ensure all resources are successfully deleted as we are left with few resources sometimes.
+          for i in {1..5}; do
+            ./kitctl delete kit-github-${os_type}-${{ github.run_id }}-${{github.run_attempt}}
```

Comment:
Deleting subtrate should happen in a separate step so that no resources are leaked even if early steps fail.
Ref: https://stackoverflow.com/questions/58858429/how-to-run-a-github-actions-step-even-if-the-previous-step-fails-while-still-f